### ISSUE TITLE
Should we check free disk size before syncing blocks

### ISSUE THREAD

**Issue by crackercracked** [bitcoin/bitcoin#15813]
This happens to me that my computer forces qt to exit because my disk is running out of memory

Should we skip syncing or check and show warning beforehand  when disk size available is enough for syncing all blocks?

Also wondering how large free disks should I prepare for syncing all the blocks?

**Comment by benthecarman** [comment]
As for how much free disk space you need, my datadir folder currently it taking up about ~300gb, this however has fully syned mainnet and testnet.  Without testnet it is about 30gb less.

**Comment by MarcoFalke** [comment]
I believe the gui intro does this, but not the bitcoind

**Comment by MarcoFalke** [comment]
I guess it could print a warning or error on init

**Comment by darosior** [comment]
I started looking into this and it is already done at the eleventh step in init : https://github.com/bitcoin/bitcoin/blob/dae72998e8576439254ee497e82e2b9ce4917977/src/init.cpp#L1681
But it uses the default minimum space for `CheckDiskSpace` of 50MiB
https://github.com/bitcoin/bitcoin/blob/2d46f1be0c3c8b7287aa1f62bb1f5b4a8d00ff6e/src/util/system.cpp#L138
we could estimate a better `additional_bytes` value depending on initial sync, pruning, etc ? But can we hardcode it ? We cannot know each block size before starting downloading them

**Comment by MarcoFalke** [comment]
Yeah, but we can assume the accumulative block size of the currently most-work valid chain. See `m_assumed_blockchain_size` (It is used that way in the gui intro)
